src/test/java/org/ethereum/beacon/discovery/integration/DiscoveryIntegrationTest.java

```diff
@@ -84,13 +84,39 @@ public void shouldNotSuccessfullyPingBootnodeWhenNodeRecordIsNotSigned() throws
     assertThrows(TimeoutException.class, () -> waitFor(pingResult));
   }
 
+  @Test
+  public void shouldHandleNodeChangingSourceAddress() throws Exception {
+    final DiscoverySystem node1 = createDiscoveryClient();
+    final ECKeyPair keyPair = Functions.generateECKeyPair();
+    final DiscoverySystem node2 = createDiscoveryClient(keyPair, node1.getLocalNodeRecord());
+
+    // Communicate on the first port
+    waitFor(node2.ping(node1.getLocalNodeRecord()));
+    node2.stop();
+
+    // Then with the same node ID but a different port
+    final DiscoverySystem node2OnDifferentPort =
+        createDiscoveryClient(keyPair, node1.getLocalNodeRecord());
+    waitFor(node2OnDifferentPort.ping(node1.getLocalNodeRecord()));
```

Comment:
Is .ping() supposed to throw an error? Otherwise, I can't see this test checking correctness.

Reply:
If we don't get the PONG back correctly it will time out.  It's not really the ideal test - I'd like to be able to assert that it went through another handshake but can't at this level.  But its still useful as a basic sanity check that things do keep working (I managed to break that at one point in the development).